### PR TITLE
add attrs_as_columns bool arg to storm.to_dataset

### LINKED ISSUE #116: remove unused line
https://github.com/tropycal/tropycal/blob/master/src/tropycal/tracks/storm.py#L578

### PR DESCRIPTION
**Pull request by raybellwaves**
A first step towards #115 also closes #116

Apologies my vscode auto linted the code. Let me know if you want me to revert it.

When you do `to_dataset` on a `tropycal.tracks.Storm` you lose the attribute information which makes it difficult to know where it came from. Pandas attributes are still WIP. This solution adds to the attributes as columns in the dataframe. That way you could append multiple storms to a single dataframe and know more information about the storms.

### CODE REVIEW

File: tests/tracks.py [hunk 1]
```diff
@@ -29,94 +30,119 @@ def assert_array_almost_equal(actual, desired, decimal=7):
...
+
+    # Convert to datatypes
     storm.to_dict()
     storm.to_xarray()
     storm.to_dataframe()
-    
-    #------------------------------------------------------------
-    
-    #Test retrieving season
+    storm.to_dataframe(attrs_as_columns=True)
```
Comment (raybellwaves) [inline tests/tracks.py:122]: New test added

File: src/tropycal/tracks/storm.py [hunk 2]
```diff
@@ -482,120 +543,148 @@ def interp(self,timeres=1,dt_window=24,dt_align='middle'):
...
-        #Return dataset
+        # Convert entire dict to a Dataset
+        ds = xr.Dataset(ds, attrs=attrs)
+
+        # Return dataset
         return ds
 
-    def to_dataframe(self):
-        
+    def to_dataframe(self, attrs_as_columns=False):
```
Comment (raybellwaves) [inline src/tropycal/tracks/storm.py:630]: new code

File: src/tropycal/tracks/storm.py [hunk 3]
```diff
@@ -482,120 +543,148 @@ def interp(self,timeres=1,dt_window=24,dt_align='middle'):
...
+        # Add every key containing a list into the dict
         keys = [k for k in self.dict.keys()]
         for key in keys:
             if isinstance(self.dict[key], list) == True:
                 ds[key] = self.dict[key]
-                    
-        #Convert entire dict to a DataFrame
+            else:
+                if attrs_as_columns:
+                    ds[key] = self.dict[key]
```
Comment (raybellwaves) [inline src/tropycal/tracks/storm.py:664]: new code

**Comment by tomerburg** [comment]
There are some conflicts with this PR right now as Storm objects now have invest handling capabilities - I'm going through these now to resolve them before merging this PR.

I do have to hope linting up soon, so in the meantime I'll remove that on my end while adding the attribute information to the `to_dataframe()` function.

**Comment by raybellwaves** [comment]
@tomerburg I believed I rebased but please check. Welcome to do this PR yourself.

code is:
```
def to_dataframe(self, attrs_as_columns=False):
...
        for key in keys:
            if isinstance(self.dict[key], list) == True:
                ds[key] = self.dict[key]
            else:
                if attrs_as_columns:
                    ds[key] = self.dict[key]
...
...

**Review by tomerburg — APPROVED** [review]
Changes all look good! Only thing missing is documentation for the added keyword argument "attrs_as_columns", but this is a minor addition I can do on my end.

**Pull request merged**